### ISSUE TITLE
Use Semver Library

### ISSUE THREAD

**Issue by nagilson** [dotnet/vscode-dotnet-runtime#2003]
Instead of manually coding all of our version parsing, we should probably migrate to the semver library.
https://github.com/dotnet/vscode-dotnet-runtime/pull/2000#discussion_r1817205415 Please see this for more information.

**Comment by agarwalishita** [comment]
Can you please assign this issue on my name?
Since it's a first contribution, I may connect with you for more information.

**Comment by nagilson** [comment]
Absolutely, I would be happy to :) This will be replacing some of the logic in the VersionUtilities and the code that references them, as well as the logic mentioned in the PR above.

**Comment by nagilson** [comment]
I need to make an urgent change regarding this, so for this task I am going to pick it up now.

**Comment by nagilson** [comment]
This isn't really a simple fix. We need to support '9.0' and '9' as versions but those are not valid semver versions
